### PR TITLE
Patch Azure One Click Deploy

### LINKED ISSUE #485: Azure custom deployment fails
Following the Azure installation instructions, and the custom deployment fails on the newuserscript step with the following output.

Top level status says "Conflict"

{
    "status": "Failed",
    "error": {
        "code": "VMExtensionProvisioningError",
        "message": "VM has reported a failure when processing extension 'newuserscript'. Error message: \"Enable failed: processing file downloads failed: failed to download file[0]: failed to download file: unexpected status code: actual=404 expected=200\"\r\n\r\nMore information on troubleshooting is available at https://aka.ms/VMExtensionCSELinuxTroubleshoot "
    }
}

### PR DESCRIPTION
**Pull request by kflansburg**
Fixes #485 

Untested.

### CODE REVIEW

**Comment by bacongobbler** [comment]
Good catch. Thanks!

Should we update `_artifactslocation` instead? That way any new scripts we add to `contrib/azure` can be included in the URL without having to keep including `contrib/azure`. I don't foresee us needing to include any scripts from other directories - the one click deploy script directory should be self-contained.

**Comment by kflansburg** [comment]
> Good catch. Thanks!
> 
> Should we update `_artifactslocation` instead? That way any new scripts we add to `contrib/azure` can be included in the URL without having to keep including `contrib/azure`. I don't foresee us needing to include any scripts from other directories - the one click deploy script directory should be self-contained.

Good idea. Done.

**Comment by kflansburg** [comment]
Hmmm, looks like some spurious errors. gonna rerun.

**Review by bacongobbler — APPROVED** [review]

**Comment by kflansburg** [comment]
@bacongobbler Were you able to test this? Would be good to confirm that there aren't any other typos.

**Comment by bacongobbler** [comment]
I haven't yet. Currently letting it run and will report back once it finishes

**Comment by bacongobbler** [comment]
Test failed with the new artifacts location. Same error as was reported in #485.

```
{
    "status": "Failed",
    "error": {
        "code": "VMExtensionProvisioningError",
        "message": "VM has reported a failure when processing extension 'newuserscript'. Error message: \"Enable failed: processing file downloads failed: failed to download file[0]: failed to download file: unexpected status code: actual=404 expected=200\"\r\n\r\nMore information on troubleshooting is available at https://aka.ms/VMExtensionCSELinuxTroubleshoot "
    }
}
```

**Comment by bacongobbler** [comment]
Testing now with the original parameters to reproduce the original issue.

File: contrib/azure/azuredeploy.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
 			"metadata": {
 				"description": "The base URI where artifacts required by this template are located. For example, if stored on a public GitHub repo, you'd use the following URI: https://raw.githubusercontent.com/deislabs/krustlet/master/."
 			},
-			"defaultValue": "https://raw.githubusercontent.com/deislabs/krustlet/master/"
+			"defaultValue": "https://raw.githubusercontent.com/deislabs/krustlet/master/contrib/azure"
```
Comment (bacongobbler) [inline contrib/azure/azuredeploy.json:10]: AHA! I see the problem now. It's missing a trailing slash in the URL.

Once you add that, the value should work. Re-testing.

**Review by bacongobbler — CHANGES_REQUESTED** [review]

File: contrib/azure/azuredeploy.json (hunk 1 above)
Comment (kflansburg) [reply contrib/azure/azuredeploy.json:10]: Ah my bad, I figured this was like `os.path.join`

Comment (bacongobbler) [reply contrib/azure/azuredeploy.json]: nah, ARM is a bit dumb at times :P

**Review by bacongobbler — APPROVED** [review]
LGTM

image

**Pull request merged**